servers/src/main/java/tachyon/worker/block/BlockMetadataManager.java

```diff
@@ -127,6 +127,12 @@ public void addTempBlockMeta(TempBlockMeta tempBlockMeta)
    */
   public void commitTempBlockMeta(TempBlockMeta tempBlockMeta)
       throws WorkerOutOfSpaceException, BlockAlreadyExistsException, BlockDoesNotExistException {
+    long blockId = tempBlockMeta.getBlockId();
+    if (hasBlockMeta(blockId)) {
```

Comment:
It's a bug in my understanding. If `tempBlockMeta` has the same id as a committed block which is in another `StorageDir`, then `dir.addBlockMeta` won't throw `BlockAlreadyExistsException`. 

The new codes enable [this test](https://github.com/amplab/tachyon/pull/2106/files#diff-7332fdb1d95b81852823dda9dec454cfL429).